test-basic/query-builder.js

```diff
@@ -2001,7 +2001,50 @@ describe('query-builder', function() {
         {'false-query':null}
         );
   });
-
+  it('should create a before-query', function(){
+   var d = new String('foo');
+   assert.deepEqual(
+        q.before('foo'),
+        {'before-query':{timestamp:d}}, 'Error in Before-Query with String input'
+        );
+  });
+  it('should create a before-query', function(){
+   var d = new String(1562178870875);
+   assert.deepEqual(
+        q.before(1562178870875),
+        {'before-query':{timestamp:d}}, 'Error in Before-Query with Number'
+        );
+  });
+  it('should create a before-query', function(){
+   var input = new Date(2019, 7, 3);
+   var d = new String(1564815600000);
```

Comment:
Would the test be easier to understand as `var d = new String(input.getTime());`?